# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.h
# Commit: b3ffe3409712
# Full Hash: b3ffe3409712f31bcf61cf93aa09bf137e45f9bc
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2025-03-29 09:19:43
# Description:
#   Bug 1951554 - Avoid racing on nsHttpTransaction::mConnInfo, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D243146
# ==============================================================================

diff -r 21cdd98a6828 -r b3ffe3409712 netwerk/protocol/http/nsHttpChannel.h
--- a/netwerk/protocol/http/nsHttpChannel.h	Fri Mar 28 17:48:02 2025 +0000
+++ b/netwerk/protocol/http/nsHttpChannel.h	Fri Mar 28 17:56:38 2025 +0000
@@ -331,9 +331,10 @@
   [[nodiscard]] nsresult DispatchTransaction(
       HttpTransactionShell* aTransWithStickyConn);
   [[nodiscard]] nsresult CallOnStartRequest();
-  [[nodiscard]] nsresult ProcessResponse();
-  void AsyncContinueProcessResponse();
-  [[nodiscard]] nsresult ContinueProcessResponse1();
+  [[nodiscard]] nsresult ProcessResponse(nsHttpConnectionInfo* aConnInfo);
+  void AsyncContinueProcessResponse(nsHttpConnectionInfo* aConnInfo);
+  [[nodiscard]] nsresult ContinueProcessResponse1(
+      nsHttpConnectionInfo* aConnInfo);
   [[nodiscard]] nsresult ContinueProcessResponse2(nsresult);
   nsresult HandleOverrideResponse();
 